### ISSUE TITLE
System.ArgumentException: 'Argument types do not match'

### ISSUE THREAD

**Issue by ristogod** [AutoMapper/AutoMapper#2393]
We just updated AutoMapper from version 6.1.1 to 6.2.0.

As soon as we did we not get **System.ArgumentException: 'Argument types do not match'** on application startup.

It is thrown from this code

``` csharp
var mapperFactory = new MapperFactory<MapperConfig>();
            mapperFactory.Add(Default,
                              new MapperConfiguration(_ =>
                                                      {
                                                          _.AddCollectionMappers();
                                                          _.AddProfile<PagedDataProfile>();
                                                          _.AddProfile<Business.DefaultProfile>();
                                                          _.AddProfile<Data.DefaultProfile>();
                                                          _.AddProfile<Services.DefaultProfile>();
                                                          _.AddProfile<DefaultProfile>();
                                                      }).CreateMapper());
```

I've tried looking up how to resolve this, but found nothing. I don't understand what's causing it. Everything worked before updating.

**Comment by lbargaoanu** [comment]
A repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by ristogod** [comment]
I have to be honest. I don't really understand how to do that. 

Also, I'm not certain the issue is reproducible outside our application. And the application can't be shared as I do not have the permissions to share it in it's entirety. 

I can try to find the time to produce something to share, but I'm not sure when that will be as my development time is shackled to the restrictive allowances of the agile development process.

I will say this though that we rolled all our code back to use 6.1.1, that being the only change and the issue disappeared. This is the only time I've encountered an issue like this with AutoMapper, where prior we've stayed current without issue.
